File: app/src/main/java/closer/vlllage/com/closer/handler/map/NetworkConnectionViewHandler.java
package closer.vlllage.com.closer.handler.map;

import android.view.View;

import closer.vlllage.com.closer.handler.helpers.ConnectionErrorHandler;
import closer.vlllage.com.closer.pool.PoolMember;

public class NetworkConnectionViewHandler extends PoolMember {

    private static final long NETWORK_CONNECTION_ERROR_TIMEOUT_MS = 5000;

    private View connectionErrorView;
    private Runnable hideCallback = new Runnable() {
        @Override
        public void run() {
            if (connectionErrorView != null) {
                connectionErrorView.setVisibility(View.GONE);
            }
        }
    };

    public void attach(View connectionErrorView) {
        this.connectionErrorView = connectionErrorView;
        $(ConnectionErrorHandler.class).setOnConnectionErrorListener(this::show);
    }

    private void show() {
        connectionErrorView.setVisibility(View.VISIBLE);
        connectionErrorView.removeCallbacks(hideCallback);
        connectionErrorView.postDelayed(hideCallback, NETWORK_CONNECTION_ERROR_TIMEOUT_MS);
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/map/NetworkConnectionViewHandler.kt
package closer.vlllage.com.closer.handler.map

import android.view.View
import closer.vlllage.com.closer.extensions.visible
import closer.vlllage.com.closer.handler.helpers.ConnectionErrorHandler
import com.queatz.on.On

class NetworkConnectionViewHandler constructor(private val on: On) {

    private lateinit var connectionErrorView: View
    private val hideCallback = Runnable {
        hide()
    }

    fun attach(connectionErrorView: View) {
        this.connectionErrorView = connectionErrorView
        on<ConnectionErrorHandler>().onConnectionErrorListener = {
            show()
        }

        this.connectionErrorView.setOnClickListener { hide() }
    }

    fun hide() {
        connectionErrorView.visible = false
    }

    private fun show() {
        connectionErrorView.visible = true
        connectionErrorView.removeCallbacks(hideCallback)
        connectionErrorView.postDelayed(hideCallback, NETWORK_CONNECTION_ERROR_TIMEOUT_MS)
    }

    companion object {

        private const val NETWORK_CONNECTION_ERROR_TIMEOUT_MS: Long = 5000
    }
}
